[Bug]: In Quarkus, build-time injection failure for unused beans from a2asdk libraries

### What happened?

I’m not sure if it’s because it wasn’t the right place, but I didn’t get a response to the [issue](https://github.com/a2aproject/a2a-samples/issues/280) I opened in a2a-samples. Unless I’m misjudging, the issue seems like a bug to me, which is why I wanted to share it again here. 

PS: The situation I mentioned also applies to the `a2a-java-reference-server` artifact.

### Relevant log output

```shell
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:3.22.3:build (default) on project weather: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: jakarta.enterprise.inject.spi.DeploymentException: Found 2 deployment problems: 
[ERROR] [1] Unsatisfied dependency for type io.a2a.spec.AgentCard and qualifiers [@PublicAgentCard]
[ERROR]         - injection target: parameter 'agentCard' of io.a2a.server.requesthandlers.JSONRPCHandler constructor
[ERROR]         - declared on CLASS bean [types=[io.a2a.server.requesthandlers.JSONRPCHandler, java.lang.Object], qualifiers=[@Default, @Any], target=io.a2a.server.requesthandlers.JSONRPCHandler]
[ERROR]         The following classes match by type, but have been skipped during discovery:
[ERROR]         - io.a2a.spec.AgentCard has no bean defining annotation (scope, stereotype, etc.)
[ERROR] 
[ERROR] 
[ERROR] [2] Unsatisfied dependency for type io.a2a.server.agentexecution.AgentExecutor and qualifiers [@Default]
[ERROR]         - injection target: parameter 'agentExecutor' of io.a2a.server.requesthandlers.DefaultRequestHandler constructor
[ERROR]         - declared on CLASS bean [types=[io.a2a.server.requesthandlers.DefaultRequestHandler, java.lang.Object, io.a2a.server.requesthandlers.RequestHandler], qualifiers=[@Default, @Any], target=io.a2a.server.requesthandlers.DefaultRequestHandler]
[ERROR]         at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:1589)
[ERROR]         at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:338)
[ERROR]         at io.quarkus.arc.processor.BeanProcessor.initialize(BeanProcessor.java:178)
[ERROR]         at io.quarkus.arc.deployment.ArcProcessor.validate(ArcProcessor.java:488)
[ERROR]         at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
[ERROR]         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:1570)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:499)
[ERROR]         Suppressed: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.a2a.spec.AgentCard and qualifiers [@PublicAgentCard]
[ERROR]         - injection target: parameter 'agentCard' of io.a2a.server.requesthandlers.JSONRPCHandler constructor
[ERROR]         - declared on CLASS bean [types=[io.a2a.server.requesthandlers.JSONRPCHandler, java.lang.Object], qualifiers=[@Default, @Any], target=io.a2a.server.requesthandlers.JSONRPCHandler]
[ERROR]         The following classes match by type, but have been skipped during discovery:
[ERROR]         - io.a2a.spec.AgentCard has no bean defining annotation (scope, stereotype, etc.)
[ERROR] 
[ERROR] 
[ERROR]                 at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:547)
[ERROR]                 at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:689)
[ERROR]                 at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:323)
[ERROR]                 ... 12 more
[ERROR]         Suppressed: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.a2a.server.agentexecution.AgentExecutor and qualifiers [@Default]
[ERROR]         - injection target: parameter 'agentExecutor' of io.a2a.server.requesthandlers.DefaultRequestHandler constructor
[ERROR]         - declared on CLASS bean [types=[io.a2a.server.requesthandlers.DefaultRequestHandler, java.lang.Object, io.a2a.server.requesthandlers.RequestHandler], qualifiers=[@Default, @Any], target=io.a2a.server.requesthandlers.DefaultRequestHandler]
[ERROR]                 at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:547)
[ERROR]                 at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:689)
[ERROR]                 at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:323)
[ERROR]                 ... 12 more
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/reference-impl/src/main/java/io/a2a/server/apps/quarkus/DefaultProducers.java
+++ b/reference-impl/src/main/java/io/a2a/server/apps/quarkus/DefaultProducers.java
@@ -1,0 +1,48 @@
+package io.a2a.server.apps.quarkus;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Default;
+import jakarta.ws.rs.Produces;
+
+import io.a2a.server.PublicAgentCard;
+import io.a2a.server.agentexecution.AgentExecutor;
+import io.a2a.server.agentexecution.RequestContext;
+import io.a2a.server.events.EventQueue;
+import io.a2a.spec.AgentCapabilities;
+import io.a2a.spec.AgentCard;
+import io.a2a.spec.JSONRPCError;
+import io.quarkus.arc.DefaultBean;
+
+/**
+ * Contains beans annotated with the Quarkus @DefaultBean annotation, in order to avoid
+ * injection failures when building the Quarkus application as discussed in
+ * <a href="https://github.com/a2aproject/a2a-java/issues/213">Issue 213</a>.
+ *
+ * If an application provides actual implementations of these beans,
+ * those will be used instead.
+ */
+@ApplicationScoped
+public class DefaultProducers {
+    @Produces
+    @PublicAgentCard
+    @DefaultBean
+    public AgentCard createDefaultAgentCard() {
+        throw new IllegalStateException(wrap("Please provide your own AgentCard implementation"));
+    }
+
+    @Produces
+    @Default
+    @DefaultBean
+    public AgentExecutor createDefaultAgentExecutor() {
+        throw new IllegalStateException(wrap("Please provide your own AgentExecutor implementation"));
+    }
+
+    private String wrap(String s) {
+        return s +
+                " as a CDI bean. Your bean will automatically take precedence over this @DefaultBean " +
+                "annotated implementation.";
+    }
+}

